Assets/Scripts/Scenes/InGame/Player/Item/Compass/CompassMove.cs

```diff
@@ -0,0 +1,97 @@
+using System.Collections;
+using System.Collections.Generic;
+using Unity.VisualScripting;
+using UnityEngine;
+using DG.Tweening;
+
+public class CompassMove : MonoBehaviour
+{
+    private GameObject _escapePoint;
+    private GameObject[] _Enemys;
+    private float distance;
+    private bool _isEnabled = true;
+    [Header("�j�̉�]���x")]
+    [SerializeField] private float _rotateSpeed;
+    [Header("�j�̉�]����(�̏᎞)")]
+    [SerializeField] private float _brokeRotateCycle;
+    [Header("�R���p�X�����鋗��")]
+    [SerializeField] private float _unusableDistance;
+    private void Start()
+    {
+        _escapePoint = GameObject.Find("escapePoint(Clone)");
+        _Enemys = GameObject.FindGameObjectsWithTag("Enemy");
+    }
+    void Update()
+    {
+
+        for (int i = 0; i < _Enemys.Length; i++)
+        {
+            Vector3 vector = _Enemys[i].transform.position - this.transform.position;
+
+
+            if (i == 0)
+            {
+                distance = vector.magnitude;
+            }
+
+            else
+            {
+                if (vector.magnitude < distance)
+                {
+                    distance = vector.magnitude;
+                }
+            }
+
+        }
+
+
+        if (distance >= _unusableDistance)
```

Comment:
```suggestion
           if (_enemy.Any(enemy => Vector3.Distance(enemy.transfrom.position,this.transform.position) >= _unusableDistance))
```
こんな感じにLinqの機能を使えばもう少しシンプルに書けそうです。
※動作確認をしていないので、もしかしたらエラーが出るかも
Linqについて → https://www.fenet.jp/dotnet/column/language/2162/